src/cmd-tag

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/python3 -u
+# Allows users to operate on the tags in `builds.json`
+#
+# Examples:
+#
+#   $ coreos-assembler tag update --tag smoketested --build 47.152
+#   $ coreos-assembler tag delete --tag beta
+#   $ coreos-assembler tag list
+#
+import argparse
+import json
+import os
+import sys
+
+sys.path.insert(0, '/usr/lib/coreos-assembler')
+from cmdlib import fatal, rfc3339_time, write_json
+
+
+def main():
+    """Main entry point"""
+
+    args = parse_args()
+    args.func(args)
+
+
+def parse_args():
+    """Parse args and dispatch"""
+
+    parser = argparse.ArgumentParser()
+    subparsers = parser.add_subparsers(dest="cmd", title="Tag actions")
+    subparsers.required = True
+
+    delete = subparsers.add_parser("delete",
+                                   help="Delete a tag from the metadata")
+    delete.add_argument("--tag", help="Tag to delete from metadata",
+                        required=True)
+    delete.set_defaults(func=cmd_delete)
+
+    list_tags = subparsers.add_parser("list", help="List available tags in "
+                                                   "the metadata")
+    list_tags.set_defaults(func=cmd_list)
+
+    update = subparsers.add_parser("update", help="Update existing tag or "
+                                                  "create a new tag")
+    update.add_argument("--tag", help="Tag to be updated", required=True)
+    update.add_argument("--build", help="Build to update tag with",
+                        required=True)
+    update.add_argument("--force", help="Force the update of a tag",
+                        action="store_true")
+    update.set_defaults(func=cmd_update)
+
+    return parser.parse_args()
+
+
+def init_build_data():
+    """Initialize build metadata"""
+
+    builds_json = "builds/builds.json"
+
+    if os.path.isfile(builds_json):
+        with open(builds_json) as json_file:
+            build_data = json.load(json_file)
+    else:
+        print("INFO: Did not find existing builds.json; starting with empty "
+              "build list and tags list")
+        build_data = {"builds": [], "tags": []}
+
+    return build_data
+
+
+def finalize_json(build_data):
+    """Wrapper around cmdlib.write_json()"""
+
+    builds_json = "builds/builds.json"
+
+    build_data['timestamp'] = rfc3339_time()
+    write_json(builds_json, build_data)
+
+
+def available_builds(build_data):
+    """Returns the availble build IDs from build metadata"""
+
+    return build_data["builds"]
```

Comment:
Is this just future-proofing?  As it stands, the function call will take more characters than this line that it's wrapping.